src/fiskaltrust.Launcher/Commands/DoctorCommand.cs

```diff
@@ -0,0 +1,434 @@
+using System.CommandLine;
+using System.CommandLine.Invocation;
+using fiskaltrust.Launcher.ProcessHost;
+using fiskaltrust.Launcher.Services;
+using Serilog;
+using ProtoBuf.Grpc.Server;
+using fiskaltrust.Launcher.Download;
+using fiskaltrust.Launcher.Extensions;
+using fiskaltrust.Launcher.Helpers;
+using fiskaltrust.Launcher.Common.Extensions;
+using fiskaltrust.Launcher.Common.Configuration;
+using fiskaltrust.Launcher.Configuration;
+using fiskaltrust.storage.serialization.V0;
+using fiskaltrust.Launcher.Services.Interfaces;
+using Grpc.Net.Client;
+using ProtoBuf.Grpc.Client;
+using fiskaltrust.Middleware.Abstractions;
+using fiskaltrust.Launcher.AssemblyLoading;
+using fiskaltrust.Launcher.Clients;
+using fiskaltrust.ifPOS.v1.de;
+using fiskaltrust.ifPOS.v1;
+
+namespace fiskaltrust.Launcher.Commands
+{
+    public class DoctorCommand : RunCommand
+    {
+        public DoctorCommand(string name = "doctor") : base(name) { }
+    }
+
+    public class DoctorCommandHandler : ICommandHandler
+    {
+        public LauncherConfiguration ArgsLauncherConfiguration { get; set; } = null!;
+        public string LauncherConfigurationFile { get; set; } = null!;
+        public string LegacyConfigurationFile { get; set; } = null!;
+        public bool MergeLegacyConfigIfExists { get; set; }
+
+        private const string SUCCESS = "✅";
+        private const string ERROR = "❌";
+        private readonly ILifetime _lifetime;
+        private readonly LauncherExecutablePath _launcherExecutablePath;
+        private bool _failed = false;
+
+        public DoctorCommandHandler(ILifetime lifetime, LauncherExecutablePath launcherExecutablePath)
+        {
+            _lifetime = lifetime;
+            _launcherExecutablePath = launcherExecutablePath;
+        }
+
+        public async Task<int> InvokeAsync(InvocationContext context)
+        {
+            try
+            {
+                Log.Logger = new LoggerConfiguration()
+                    .AddLoggingConfiguration()
+                    .CreateLogger();
+
+                LauncherConfiguration launcherConfiguration = new();
+
+                if (File.Exists(LauncherConfigurationFile))
```

Comment:
Is this now done in multiple commands? Might make sense to extract it to a method 🤔

Reply:
Talking about the launcher config stuff 😅

Reply:
It is normally done in the `Common` command. I've extracted it here because we dont need all of the other logic that does there 🤔